dandi/girder.py

```diff
@@ -353,14 +353,20 @@ def get_download_file_iter(self, file_id, chunk_size=MAX_CHUNK_SIZE):
         """
         """
 
-        def downloader():
+        def downloader(start_at=0):
             # TODO: make it a common decorator here?
             # Will do 3 attempts to avoid some problems due to flaky/overloaded
             # connections, see https://github.com/dandi/dandi-cli/issues/87
             for attempt in range(3):
                 try:
-                    return self.downloadFileAsIterator(file_id, chunkSize=chunk_size)
-                    break
+                    path = f"file/{file_id}/download"
+                    headers = None
+                    if start_at > 0:
+                        headers = {"Range": f"bytes={start_at}-"}
```

Comment:
oh man -- that would work?! ;)  how could we test I wonder...

Reply:
According to [the SO question you linked](https://stackoverflow.com/questions/22894211/how-to-resume-file-download-in-python/22894873#22894873), that works.  However, not all servers support ranges to begin with.  It's possible to test for support by checking the response for an "Accept-Ranges" header (cf. [this MDN article](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests), which doesn't mention the "start-" syntax).

I just noticed something in that MDN article: The response to a "Range" request has status 206, but `girder_client` regards anything other than 200 and 201 as an error ([see code here](https://girder.readthedocs.io/en/latest/_modules/girder_client.html#GirderClient.sendRestRequest)).

Reply:
Thanks for the detailed analysis!  I guess for `girder` lets
- submit a PR  (that could be helpful to seek feedback from its developers -- may be it couldn't work in principle with it?)
- since it would take time I bet to get PR merged and new girder-client released, we could shim `sendRestRequest` by overloading it, and if raised `HttpError` status is 206, redo handling of the `.response` which seems to be stored in the exception in its entirety.  It is ugly somewhat but I do not see why it should not work. Then we could make shim conditional on the version of girder_client.

Reply:
PR submitted: https://github.com/girder/girder/pull/3301

Reply:
`sendRestRequest()` overloaded in commit 89f84d7.